Objects/typeobject.c

```diff
@@ -3827,6 +3827,22 @@ type_new_impl(type_new_ctx *ctx)
     // Put the proper slots in place
     fixup_slot_dispatchers(type);
 
+    if (!_PyDict_HasOnlyStringKeys(type->tp_dict)) {
+        PyObject *name_of_class = type_name(type, NULL);
+        if (name_of_class == NULL) {
+            return NULL;
+        }
+        if (PyErr_WarnFormat(NULL,
```

Comment:
It is very uncommon to use this API with NULL as category. It would be less confusing to specify the category explicitly.